docs/topics/security.txt

```diff
@@ -5,70 +5,97 @@ Security in Django
+    particularly when assessing individual points in security audits.
+
 .. _cross-site-scripting:
 
-Cross site scripting (XSS) protection
+Cross-site scripting (XSS) protection
 =====================================
 
 .. highlightlang:: html+django
 
-XSS attacks allow a user to inject client side scripts into the browsers of
-other users. This is usually achieved by storing the malicious scripts in the
-database where it will be retrieved and displayed to other users, or by getting
-users to click a link which will cause the attacker's JavaScript to be executed
-by the user's browser. However, XSS attacks can originate from any untrusted
-source of data, such as cookies or Web services, whenever the data is not
-sufficiently sanitized before including in a page.
+In a cross-site scripting attack, malicious code in the form of a client-side
+script is injected into another user's web browser, where it will be executed.
+
+This is typically done by:
+
+* storing the malicious script in the database where it will be retrieved and
+  presented to other users in their browsers, or
+* getting users to click a link which will cause the attacker's JavaScript to
+  be executed by the user's browser.
+
+Cross-site scripting attacks can originate from any untrusted source of data,
+including cookies or web services, if the data are not adequately sanitized
+before being published in a page.
 
-Using Django templates protects you against the majority of XSS attacks.
-However, it is important to understand what protections it provides
-and its limitations.
+Django templates provide protection against the majority of cross-site
+scripting attacks by :ref:`automatically escaping characters that represent a
+risk <automatic-html-escaping>` (that is, HTML characters that could be
+*interpreted* by the browser to malicious effect are instead safely
+*displayed*).
 
-Django templates :ref:`escape specific characters <automatic-html-escaping>`
-which are particularly dangerous to HTML. While this protects users from most
-malicious input, it is not entirely foolproof. For example, it will not
-protect the following:
+However, the extent of this protection and its limitations should be understood.
+
+For example, consider the case below, where the template expects ``var`` to
+contain something like ``'class1'``:
 
 .. code-block:: text
 
     <style class={{ var }}>...</style>
 
 .. highlighting as html+django fails due to intentionally missing quotes.
 
-If ``var`` is set to ``'class1 onmouseover=javascript:func()'``, this can result
-in unauthorized JavaScript execution, depending on how the browser renders
-imperfect HTML. (Quoting the attribute value would fix this case.)
+If ``var`` is set to ``'class1 onmouseover=javascript:func()'``, this could
+result in unauthorized JavaScript execution, depending on how the browser
+renders imperfect HTML. Better template design, in which ``{{ var }}`` is
+quoted in the template itself, would fix this.
```

Comment:
Could we provide an example of how this would look as an adjustment to the above example?